index.js

```diff
@@ -0,0 +1,231 @@
+"use strict";
+
+class Order {
+
+  products = [];
+  isPaid = false;
+
+  getTotal() {
+    const productsCopy = this.products.slice();
+    const total = productsCopy.reduce((currentTotal, product) => {
+      return currentTotal += product.price;
+    }, 0);
+    return total;
+  };
+
+  getCalories() {
+    const productsCopy = this.products.slice();
+    const calories = productsCopy.reduce((currentCalories, product) => {
+      return currentCalories += product.calories;
+    }, 0);
+    return calories;
+  };
+
+  addProduct(product) {
+    if (this.isPaid) {
+      console.log("The order is already paid. No changes allowed.");
+      return;
+    }
+    this.products.push(product);
+    return this;
+  };
+
+  deleteProduct(product) {
+    if (this.isPaid) {
+      console.log("The order is already paid. No changes allowed.");
+      return;
+    }
+    if (this.products.includes(product)) {
+      this.products = this.products.filter(item => item !== product);
+    } else {
+      console.log("The product is not found in the order");
+    }
+    return this;
+  };
+
+  payOrder() {
+    this.isPaid = true;
+    console.log("The order is paid");
+    return this;
+  };
+
+};
+
+class Product {
+
+  constructor(name) {
+    this.name = name;
+    this.price = this.getPrice();
+    this.calories = this.getCalories();
+    this.id = this.getId();
+  }
+
+  getPrice() { };
+  getCalories() { };
+  getId() {
+    return Math.round(Math.random() * 10000);
+  }
+
+}
+
+class Salad extends Product {
+
+  static CAESAR = "caesar";
+  static OLIVIE = "olivie";
+  static CATEGORY = "salads";
+  static STANDARD_SERVING = 100;
+
+  constructor(name = Salad.CAESAR, quantity = Salad.STANDARD_SERVING) {
+    super(name);
+    this.category = Salad.CATEGORY;
+    this.quantity = isFinite(quantity) && quantity > 0 ? quantity : Salad.STANDARD_SERVING;
+    this.price = this.getPrice();
+    this.calories = this.getCalories();
+  }
+
+  getPrice() {
```

Comment:
What if you have lots of salad type? More than 10? The method should be universal and shoundn't depend on types of the products. It's better to change logic of all methods which use `if` logic and depend on certain project.